### PR TITLE
Support Decimal values in JSONEncoder

### LINKED ISSUE #412: JSON encoding of Decimal fails
### Description

Decimal values could be returned from DB queries.
This issue could be fixed by patching ``app.json_encoder``, but I thought it could be useful to be part of the base connexion encoder.

```python
TypeError: Decimal('0') is not JSON serializable
```

### Expected behaviour

API responds with valid JSON including float values (e.g. ``0.1`` or ``1.1e-05`` etc ...)

### Actual behaviour

JSON serialization fail

### Steps to reproduce

```python
import json
from decimal import Decimal
from connexion.decorators.produces import JSONEncoder

json.dumps(Decimal(0.1), cls=JSONEncoder)
```

### Additional info:
http://www.json.org/
http://stackoverflow.com/questions/19554972/json-standard-floating-point-numbers

Output of the commands:

- `python --version`
``Python 3.5.2``

- `pip show connexion | grep "^Version\:"`
``Version: 1.1.4``

### PR DESCRIPTION
**Pull request by mohabusama**
Fixes #412 .

Changes proposed in this pull request:

 - Update JSONEncoder to handle ``Decimal`` values.

### CODE REVIEW

File: connexion/decorators/produces.py [hunk 1]
```diff
@@ -30,6 +32,9 @@ def default(self, o):
         if isinstance(o, datetime.date):
             return o.isoformat()
 
+        if isinstance(o, Decimal):
+            return float(o)
```
Comment (hjacobs) [inline connexion/decorators/produces.py]: but this would lose precision..

**Comment by hjacobs** [comment]
Can you check the Travis build?

File: connexion/decorators/produces.py (hunk 1 above)
Comment (mohabusama) [inline connexion/decorators/produces.py]: We can fallback to strings for precise representation, but I am not sure if this is the desired ``default`` behavior. I felt using floats (👿 ) is somehow expected/acceptable, unless precision is required which in this case dev would use their own implementation.

**Comment by mohabusama** [comment]
Build looks fine after a restart.

**Comment by hjacobs** [comment]
Thanks for the PR!

**Comment by hjacobs** [comment]
:+1:

**Comment by hjacobs** [comment]
:+1:

**Pull request merged**